### PR TITLE
Add ml_model function wrapper API

### PR DESCRIPTION
**Pull request by umaannamalai**
This PR adds a new API that allows users to pass information about their sklearn models including feature names, model names/ versions, and label names.

### CODE REVIEW

**Review by hmstepanek — COMMENTED** [review]
Just posting this as is based on our discussion.

File: newrelic/hooks/mlmodel_sklearn.py [hunk 1]
```diff
@@ -105,26 +106,36 @@ def wrap_predict(transaction, _class, wrapped, instance, args, kwargs):
...
+                    column_name_mapping = dict(zip(data_set.columns.tolist(), feature_names))
+
             for (colname, colval) in data_set.iteritems():
                 for value in colval.values:
                     value_type = data_set[colname].dtype.name
                     if value_type == "category":
                         value_type = "categorical"
                     else:
                         value_type = find_type_category(value)
+                    feature_name = column_name_mapping[colname] if colname in column_name_mapping else colname
```
[the remark this answers is not in the document]
Comment (hmstepanek) [reply newrelic/hooks/mlmodel_sklearn.py:131]: Just as a reminder. The below is 3 predictions with 2 feature columns.
```
>>> np.array(pd.DataFrame({"col1": [0,1,2], "col2":[3,4,5]}))
array([[0, 3],
       [1, 4],
       [2, 5]])
```
The user may only specify one prediction. In which case I think we could cast this to a 2D array:
```
>>> x = [0,1]
>>> np.array([x])
array([[0, 1]])
```

[the remark this answers is not in the document]
Comment (hmstepanek) [reply newrelic/hooks/mlmodel_sklearn.py:131]: If it comes in handy at all: [newrelic/hooks/mlmodel_sklearn.py](https://github.com/newrelic/newrelic-python-agent/blob/27ad2976a3459b48c5f4b32812a85823e2d82e6d/newrelic/hooks/mlmodel_sklearn.py) in `_calc_prediction_feature_stats`

File: tests/mlmodel_sklearn/test_ml_model.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+            "model_name": "MyDecisionTreeClassifier",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature2",
+            "type": "numerical",
+            "value": "7.5",
+        }
+    },
+]
+
+
```
Comment (hmstepanek) [inline tests/mlmodel_sklearn/test_ml_model.py:302]: We need a couple tests for the error/warning cases:
* if the feature names don't match the number of feature columns and it's a pandas dataframe
* if the feature names don't match the number of feature columns and it's a np array

**Review by hmstepanek — CHANGES_REQUESTED** [review]
Looking much better! I had a couple implementation suggestions to simplify things/fix some issues. I also requested a couple more tests.

File: tests/mlmodel_sklearn/test_ml_model.py [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+try:
+    from sklearn.tree._classes import BaseDecisionTree
+except ImportError:
+    from sklearn.tree.tree import BaseDecisionTree
+
+_logger = logging.getLogger(__name__)
+
+
+class CustomTestModel(BaseDecisionTree):
```
[the remark this answers is not in the document]
Comment (umaannamalai) [reply tests/mlmodel_sklearn/test_ml_model.py:36]: Yes, that is true! Is there a way to simplify the __init__?

**Review by hmstepanek — APPROVED** [review]
Looks good! Thanks for bearing with me on this-it's been a lot of back and forth! I'm gonna pre-approve this one. If the args,kwargs thing works for implementing the init, great, if not feel free to merge as is.
